darwin/future/meta/objects/dataset.py

```diff
@@ -0,0 +1,67 @@
+from typing import List, Optional, Tuple, Union
+
+from darwin.future.core.datasets.get_dataset import get_dataset
+from darwin.future.core.datasets.remove_dataset import remove_dataset
+from darwin.future.data_objects.dataset import Dataset
+from darwin.future.meta.client import MetaClient
+from darwin.future.meta.queries.dataset import DatasetQuery
+
+
+class DatasetMeta:
+    client: MetaClient
+
+    def __init__(self, client: MetaClient) -> None:
+        # TODO: Initialise from chaining within MetaClient
+        self.client = client
+
+    def datasets(self) -> DatasetQuery:
```

Comment:
Does this make sense? Having a dataset object that returns a query for datasets.

Reply:
The idea I had in mind was that it would be for later, and would basically be refactored, when we have the chaining like this:

```python

client.datasets().where(etc...)
```

So it's a placeholder for now, the idea being that it might be useful in wiring it all up later.

I've got a vague idea in mind of using the `__getattribute__` dunder to lazy load these on `MetaClient`, but ATM that's just a blue sky thought.  Could allow invocation without invocation though.